lib/neo4j/http/node_client.rb

```diff
@@ -65,6 +67,20 @@ def build_selectors(attributes, node_name: :node)
           end
         end.join(" AND ")
       end
+
+      def upsert_node_via_unwind(node, unwind)
+        cypher = <<-CYPHER
+          UNWIND $batch as row
+          MERGE (node:#{node.label} {#{node.key_name}: row.#{node.key_name}})
+          ON CREATE SET node += row
+          ON MATCH SET node += row
+          return node
+        CYPHER
+
+        results = @cypher_client.execute_cypher(cypher, key_value: node.key_value, batch: unwind)
```

Comment:
unwind is an array of node properties?

Reply:
will the whole thing fail if a schema violation happens?

Reply:
@Tiphani unwind would be an array of Hash objects e.g.
```
[
          {
            from: {
              uuid: "FromUuid"
            },
            to: {
              uuid: "ToUuid"
            },
            relationship: {
              uuid: "RelationshipUuid",
              age: 33
            }
          }
        ]
```

Reply:
As for a schema violation failure, the entire transaction will rollback. I've now added this as a spec